clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -409,6 +409,9 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     update_switch_stats(output_ordinal_t output, input_ordinal_t prev_input,
                         input_ordinal_t new_input);
 
+    void
+    process_marker(RecordType record, output_ordinal_t output);
```

Comment:
schedule_dynamic_t has a virtual method of the same name: so this may cause confusion.  Seems best to make this the same name and signature and virtual and override in schedule_dynamic_t. 
 (Maybe an alternative is to give this a different name.)

Reply:
>  Seems best to make this the same name and signature and virtual and override in schedule_dynamic_t.

I updated the name to update_syscall_state.

Reply:
> Seems best to make this the same name and signature and virtual and override in schedule_dynamic_t.

Also added a comment to clarify that we want check_for_input_switch to see the updated state, so we must process the syscall trace related markers now.

Reply:
I have a question: do we consider the injected sequences to be associated with the input streams?

For switch sequences, maybe not.

For syscall sequences, likely yes? If so, is it prudent to maintain state like `in_syscall_code` on the output_info_t? What if scheduler_dynamic_tmpl_t decides it wants to do a context switch in the middle of syscall code (though today we have logic that prevents switches while in syscall code): state like in_syscall_code will become invalid if the output switches to a different input. Maybe in_syscall_code should be moved to input_info_t?

Reply:
Just moving in_syscall_code does seem reasonable.  I assume you're not proposing counting the injected sequence as part of the input stream.

Reply:
> I assume you're not proposing counting the injected sequence as part of the input stream.

Nope, not suggesting that.

> Just moving in_syscall_code does seem reasonable.

Added an XXX comment for now. This PR is already too big.